pkg/conjurpolicy/types.go

```diff
@@ -185,60 +35,118 @@ type User struct {
 	Annotations Annotations `yaml:"annotations,omitempty"`
 }
 
+type Policy struct {
+	Resource    `yaml:"-"`
+	Id          string           `yaml:"id"`
+	Annotations Annotations      `yaml:"annotations,omitempty"`
+	Owner       ResourceRef      `yaml:"owner,omitempty"`
+	Body        PolicyStatements `yaml:"body,omitempty"`
+}
+
+type Layer struct {
+	Resource `yaml:"-"`
+}
+
+type Grant struct {
+	Resource `yaml:"-"`
+	Role     ResourceRef `yaml:"role"`
+	Member   ResourceRef `yaml:"member"`
+}
+
+type Host struct {
+	Resource    `yaml:"-"`
+	Id          string            `yaml:"id,omitempty"`
+	Owner       ResourceRef       `yaml:"owner,omitempty"`
+	Body        PolicyStatements  `yaml:"body,omitempty"`
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+type Delete struct {
+	Resource `yaml:"-"`
+	Record   ResourceRef `yaml:"record"`
+}
+
+type Permit struct {
+	Resource   `yaml:"-"`
+	Role       ResourceRef `yaml:"role"`
+	Privileges []Privilege `yaml:"privileges,flow"`
+	Resources  ResourceRef `yaml:"resource"`
+}
+
+type Deny struct {
+	Resource   `yaml:"-"`
+	Role       ResourceRef `yaml:"role"`
```

Comment:
TODO: Probably something to be considered separate from this PR... how can we lock this down to role refs only